src/StatPlots.jl

```diff
@@ -17,7 +17,9 @@ import Loess
 
 export groupapply
```

Comment:
Agreed on both.
I also think `groupapply` can benefit from this (and also from the adjustments to the `group` keyword in [#1033](https://github.com/JuliaPlots/Plots.jl/pull/1033) ) but I need to think carefully about it. It probably deserves its own issue.

In terms of syntax, one idea would be to also allow:
```julia
@given df groupapply(...) plot(; kwargs....)
```
to mean the same as
```julia
plot(@given df groupapply(...); kwargs....)
```
But I guess we should open a separate issue to discuss changes to `groupapply`